### PR TITLE
Fixes #37452 - Support Zeitwerk loader

### PR DESCRIPTION
**Pull request by ofedoren**
~WIP~

### CODE REVIEW

File: lib/foreman_tasks/engine.rb [hunk 1]
```diff
@@ -186,6 +185,16 @@ class Engine < ::Rails::Engine
       ::Dynflow::ActiveJob::QueueAdapters::JobWrapper.include Actions::TaskSynchronization
     end
 
+    initializer "foreman_tasks.observable_events" do
```
Comment (ofedoren) [inline lib/foreman_tasks/engine.rb:188]: Can be dropped from this PR.

That's my attempt to fix at least some `autoloadable constant loading during initialization` warnings. Seems I was able to fix some of them in the plain Foreman, but with each plugin the number of warnings rises exponentially. Not required for Zeitwerk, but in the future Rails will make these warnings be errors.

**Review by ekohl — COMMENTED** [review]
Should we make `ID` an acronym to avoid renaming?

File: lib/foreman_tasks/engine.rb (hunk 1 above)
Comment (ekohl) [reply lib/foreman_tasks/engine.rb:188]: Is that something we can already merge today or wouldn't it make sense without Zeitwerk?

**Comment by ofedoren** [comment]
> Should we make `ID` an acronym to avoid renaming?

We could, but it will imply that we would need to either make custom inflector for the plugin, which won't affect core and other plugins or make it acronym in the core, so we have consistency. The later will bring even more changes across the plugins :/

File: lib/foreman_tasks/engine.rb (hunk 1 above)
Comment (ofedoren) [reply lib/foreman_tasks/engine.rb:188]: That's experimental and I'll drop this from the PR.

I think we could deal with the warnings without Zeitwerk enabled, but the warnings appear only if we enable it. I'd postpone that after we enable it for all the plugins, maybe devs will get more interested in resolving that with us.

**Comment by ofedoren** [comment]
Ready to merge, dropped the commit for testing with Foreman PR, the latest run was green:
Screenshot_20240903_143351

**Review by adamruzicka — APPROVED** [review]

**Comment by adamruzicka** [comment]
Thank you @ofedoren !

**Pull request merged**